src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -666,13 +670,21 @@
   (-plan->sql [query]
     (let [has-where? (boolean (:where query))
           has-projections? (not (empty? (:projected-fields query)))
-          update-when (fn [m pred ks f]
-                        (if pred
-                          (update-in m ks f)
-                          m))
+          update-hash-fields
```

Comment:
Huh. If we had this working in the query-rec's, we wouldn't need this, as it would be baked into the 'view'.